lib/codecs/src/encoding/format/gelf.rs

```diff
@@ -322,6 +338,47 @@ mod tests {
         }
     }
 
+    #[test]
+    fn gelf_serializing_timestamp() {
+        // floating point in case of sub second timestamp
+        {
+            let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(61, 1_000_000), Utc);
+
+            let event_fields = btreemap! {
+                VERSION => "1.1",
+                SHORT_MESSAGE => "Some message",
+                HOST => "example.org",
+                TIMESTAMP => dt,
+            };
+
+            let jsn = do_serialize(true, event_fields).unwrap();
+            assert_eq!(true, jsn.get(TIMESTAMP).unwrap().is_f64());
+            assert_eq!(
+                jsn.get(TIMESTAMP).unwrap().as_f64().unwrap(),
+                dt.timestamp_millis() as f64 / 1000.0
```

Comment:
It might be a good idea to use a fixed expected value rather than using the same logic in the test as is used in the source.

Reply:
Agreed